### ISSUE TITLE
Various error report

### ISSUE THREAD

**Issue by SVS002** [voidpenguin-28/Textractor-ExtraExtensions#3]
in VndbCharNameMapper ,i found the some case the extension mistake take the name in another vndb and apply this mapping for mistake visual novel.

![Screenshot 2024-06-22 204941](https://github.com/voidpenguin-28/Textractor-ExtraExtensions/assets/147943436/5a815b5a-3452-4fd2-8906-b1cbcae709f3)
![Screenshot 2024-06-22 205052](https://github.com/voidpenguin-28/Textractor-ExtraExtensions/assets/147943436/b03b060b-40d4-4575-95ad-10efde3583c4)

in image is the game in v49438 , i have checked the raw sentence in another thread and see the thread i am using to translate. i found have the word of this game is same the name of another character in v9641 and this has applied this mapping name.

So i am thinking the extension may confuse between visual novel, if can, can you make ini file character in same file game putting. If can't, i have an idea,the textactor have saved game file of it,can u using this:
![image](https://github.com/voidpenguin-28/Textractor-ExtraExtensions/assets/147943436/a147149c-535c-4991-a15b-41fc88254d16)
make mapper file in format v?? = (the name of game the texactor are using)

**Comment by voidpenguin-28** [comment]
Hi, can you share your ini config for VndbCharNameMapper?

**Comment by SVS002** [comment]
> Hi, can you share your ini config for VndbCharNameMapper?

![image](https://github.com/voidpenguin-28/Textractor-ExtraExtensions/assets/147943436/28fb8d2f-e14b-4e2e-acdf-4327f6b38941)

**Comment by voidpenguin-28** [comment]
Thank you. The reason this is happening is because you have the VnIds for both visual novels assigned.  This makes the extension use mappings from both visual novels.

To resolve this you will need to use 'VnIds=v9641' when you are reading v9641, and 'VnIds=v49438' when you are reading v49438.

This extension has no way to distinguish between different VnIds when running a visual novel.

**Comment by voidpenguin-28** [comment]
This extension has been updated to v1.2.0, which includes a new feature that should address some of your concerns here.
https://github.com/voidpenguin-28/Textractor-ExtraExtensions/releases/tag/VndbCharNameMapper-v1.2.0

This update introduces the capability to add App Name to VnId mappings to the *VnIds* config value. In other words, specific app names can be mapped to specific vnids.

TL;DR: With this new update, if you apply the following to the VnIds config value, it would resolve your above concern. You can add the app name for 'v9641' into there as well.
```ini
VnIds="v9641|kss=v49438"
```

More detailed information on this feature below.

----
- You can define key-value pairs which map a specific app to a specific VnId.
	- This can be useful if you have multiple VnIds defined but want a specific app to use a specific VnId.
- The key would be the app name, the value would be the VnId
	- Key value pairs are delimited by an equal sign ('=')
	- The app name is the name of the executable used to run the visual novel. The easiest way to find this name is to look at the app name dropdown in Textractor.
	- ![appname-example](https://github.com/voidpenguin-28/Textractor-ExtraExtensions/assets/155984655/af6a3201-a086-40a3-a4e0-a5a2052c5302)
		- "Farthest2015" is the app name in the above example.
		- **Note: The file extension (ex: ".exe") should not be included in the app name**
	- It is okay to mix together VnIds with app name mappings and VnIds without app name mappings
		- Ex: "*ABC123=v1234|v1278|DEF456=v1188*"
- Example: 
	```ini
	VnIds="ABC123=v1234|Farthest2015=v1278|v1188"
	VnIdDelim=|
	```
- How mappings are applied:
	- If an app mapping is found for the current app, then the VnId tied to that app name will be used for name mappings.
	- If no app mappings if found for the current app, then all **non-mapped** VnIds will be used for name mappings.
		- Ex: If *VnIds="Farthest2014=v1234|ABC123=v1278|v1188|v1199"*, and the current app is named "MoeMoe". Then only VnIds v1188 & v1199 will be used for name mappings.

**Comment by SVS002** [comment]
>     * Ex: If _VnIds="Farthest2014=v1234|ABC123=v1278|v1188|v1199"_, and the current app is named "MoeMoe". Then only VnIds v1188 & v1199 will be used for name mappings.

i thought the " will be removed after each times reopen textactor so is it necessary to add \ before " ?


Ps: if can combine both vndb mapping and api, the problem of minchar may don't isn't problem.

**Comment by voidpenguin-28** [comment]
In this case, the double quotes are fine because they are just being used to wrap around the value text, so no need to add \

I am working on the api integration with this extension, but it will time some time as I take care of other priorities.

**Comment by SVS002** [comment]
> This extension has been updated to v1.2.0, which includes a new feature that should address some of your concerns here. https://github.com/voidpenguin-28/Textractor-ExtraExtensions/releases/tag/VndbCharNameMapper-v1.2.0
> 
> This update introduces the capability to add App Name to VnId mappings to the _VnIds_ config value. In other words, specific app names can be mapped to specific vnids.
> 
> TL;DR: With this new update, if you apply the following to the VnIds config value, it would resolve your above concern. You can add the app name for 'v9641' into there as well.
> 
> ```ini
> VnIds="v9641|kss=v49438"
> ```
> 
> More detailed information on this feature below.
> 
> * You can define key-value pairs which map a specific app to a specific VnId.
>   
>   * This can be useful if you have multiple VnIds defined but want a specific app to use a specific VnId.
> * The key would be the app name, the value would be the VnId
>   
>   * Key value pairs are delimited by an equal sign ('=')
>   * The app name is the name of the executable used to run the visual novel. The easiest way to find this name is to look at the app name dropdown in Textractor.
>   * ![appname-example](https://private-user-images.githubusercontent.com/155984655/342798373-af6a3201-a086-40a3-a4e0-a5a2052c5302.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk1NzQ3MjAsIm5iZiI6MTcxOTU3NDQyMCwicGF0aCI6Ii8xNTU5ODQ2NTUvMzQyNzk4MzczLWFmNmEzMjAxLWEwODYtNDBhMy1hNGUwLWE1YTIwNTJjNTMwMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyOFQxMTMzNDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMzYxMjE1ZDQ3YzA1N2Y5NTRlYjdjMjVmN2MwZjA2NjU4MmU0ZmIzZTgyZjQzY2ExYTQxNjc1MDExZTQ1ZWIxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.FazFVUabc1z0cJl4z3udUU5r3gujjovOW380nHllsVc)
>     
>     * "Farthest2015" is the app name in the above example.
>     * **Note: The file extension (ex: ".exe") should not be included in the app name**
>   * It is okay to mix together VnIds with app name mappings and VnIds without app name mappings
>     
>     * Ex: "_ABC123=v1234|v1278|DEF456=v1188_"
> * Example:
>   ```ini
>    VnIds="ABC123=v1234|Farthest2015=v1278|v1188"
>    VnIdDelim=|
>   ```
> * How mappings are applied:
>   
>   * If an app mapping is found for the current app, then the VnId tied to that app name will be used for name mappings.
>   * If no app mappings if found for the current app, then all **non-mapped** VnIds will be used for name mappings.
>     
>     * Ex: If _VnIds="Farthest2014=v1234|ABC123=v1278|v1188|v1199"_, and the current app is named "MoeMoe". Then only VnIds v1188 & v1199 will be used for name mappings.

Some game start with jp text. reopen it make the text in ini change
![image](https://github.com/voidpenguin-28/Textractor-ExtraExtensions/assets/147943436/34217796-69ae-4c14-9e12-a84887cef4f8)

**Comment by voidpenguin-28** [comment]
This is now fixed in v1.2.1:
https://github.com/voidpenguin-28/Textractor-ExtraExtensions/releases/tag/VndbCharNameMapper-v1.2.1

Note, the hex values (ex: *\xe7*) will still appear in the ini file (since that is controlled by the Textractor app), but the extension will now properly read those hex values as Japanese characters.

**Comment by SVS002** [comment]
> This is now fixed in v1.2.1: https://github.com/voidpenguin-28/Textractor-ExtraExtensions/releases/tag/VndbCharNameMapper-v1.2.1
> 
> Note, the hex values (ex: _\xe7_) will still appear in the ini file (since that is controlled by the Textractor app), but the extension will now properly read those hex values as Japanese characters.

Can u make it for another extension like api because some time,we give the system a jp word and said it translate or keep the way calling when translate.


![Screenshot 2024-06-27 233231](https://github.com/voidpenguin-28/Textractor-ExtraExtensions/assets/147943436/e3502252-139a-4204-a418-63d444ee946f)
Is it this error caused by my lap?

**Comment by voidpenguin-28** [comment]
Is that error caused by the vndb extension or the gpt extension?

**Comment by SVS002** [comment]
> Is that error caused by the vndb extension or the gpt extension?

honest i don't identify what thing caused.

**Comment by voidpenguin-28** [comment]
Can you test the vndb extension without the gpt extension to see if the error happens? If it doesn't happen, try testing the gpt extension without the vndb extension.

**Comment by SVS002** [comment]
> Can you test the vndb extension without the gpt extension to see if the error happens? If it doesn't happen, try testing the gpt extension without the vndb extension.

Sorry i know why it happen, some time text too much and the api can't load it so it cause that error

**Comment by voidpenguin-28** [comment]
Are you running into this error often?

**Comment by voidpenguin-28** [comment]
When you get the chance, test out this beta release and let me know if you get any errors. This makes a slight adjustment to the previous release which could maybe solve this issue. 
[VndbCharNameMapper-v1.2.1b.zip](https://github.com/user-attachments/files/16034901/VndbCharNameMapper-v1.2.1b.zip)

**Comment by SVS002** [comment]
> Are you running into this error often?

No but if i mistake press skip and make the so much text load, it will cause it but i don't think it is too much usually.(vndb don't cause it, only api)

**Comment by voidpenguin-28** [comment]
Oh ok, thanks for the update. Yes, it's probably related to processing too much text at once.

Adjusting the **HistorySoftCharLimit** & **MsgCharLimit** may reduce this issue.

Since I'm in the middle of a working on updates for the gpt extension, the hex-coding fix will be applied in the next release for that extension.

**Comment by SVS002** [comment]
> Oh ok, thanks for the update. Yes, it's probably related to processing too much text at once.
> 
> Adjusting the **HistorySoftCharLimit** & **MsgCharLimit** may reduce this issue.
> 
> Since I'm in the middle of a working on updates for the gpt extension, the hex-coding fix will be applied in the next release for that extension.

i meet that error,do you know why?
![image](https://github.com/voidpenguin-28/Textractor-ExtraExtensions/assets/147943436/4201c88f-b5c2-4792-9708-69f4c00df4f4)

**Comment by voidpenguin-28** [comment]
Which version of the extension are you using? There was an update 2 days ago, which I recommend trying out.

**Comment by SVS002** [comment]
> Which version of the extension are you using? There was an update 2 days ago, which I recommend trying out.

the lastest version, it hapeen when i read through over 10, i can't count how much but it hapen when after send enough times.

**Comment by voidpenguin-28** [comment]
Thank you. Looking into it.

**Comment by voidpenguin-28** [comment]
Figured it out. It's caused by the Gemini API whenever it returns a quota reached error (when you send more requests than the free version allows per minute).
```json
{
  "error": {
    "code": 429,
    "message": "Resource has been exhausted (e.g. check quota).",
    "status": "RESOURCE_EXHAUSTED"
  }
}
```

A fix to prevent this from crashing the app will be available in the next release. Finishing up a few more changes beforehand.

**Comment by voidpenguin-28** [comment]
This issue should now be fixed in v1.3.0. As stated in the other Textractor thread, the name/gender mapping feature has been implemented.
https://github.com/voidpenguin-28/Textractor-ExtraExtensions/releases/tag/GptApiTranslate-v1.3.0

**Comment by voidpenguin-28** [comment]
Closing thread, since these concerns have been resolved. Feel free to open a new thread for any future issues.
